### PR TITLE
[Merged by Bors] - feat: `a * a⁻¹ ≤ 1`

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/Field/Defs.lean [hunk 1]
```diff
@@ -35,3 +35,8 @@ class LinearOrderedField (α : Type*) extends LinearOrderedCommRing α, Field α
 instance (priority := 100) LinearOrderedField.toLinearOrderedSemifield [LinearOrderedField α] :
     LinearOrderedSemifield α :=
   { LinearOrderedRing.toLinearOrderedSemiring, ‹LinearOrderedField α› with }
+
+variable [LinearOrderedSemifield α] {a : α}
+
+lemma mul_inv_le_one : a * a⁻¹ ≤ 1 := by obtain rfl | ha := eq_or_ne a 0 <;> simp [*]
+lemma inv_mul_le_one : a⁻¹ * a ≤ 1 := by obtain rfl | ha := eq_or_ne a 0 <;> simp [*]
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Field/Defs.lean:42]: Do you think it makes sense to have the `cancel` versions we have for equalities here too?

A docstring for each of these referencing the junk value and the other lemma would be good too.

Comment (YaelDillies) [reply Mathlib/Algebra/Order/Field/Defs.lean:42]: Hmm, that's a good idea. Will add

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Algebra/Order/Field/Defs.lean [hunk 2]
```diff
@@ -35,3 +35,27 @@ class LinearOrderedField (α : Type*) extends LinearOrderedCommRing α, Field α
...
+variable [LinearOrderedSemifield α] {a b : α}
+
+/-- Equality holds when `a ≠ 0`. See `mul_inv_cancel`. -/
+lemma mul_inv_le_one : a * a⁻¹ ≤ 1 := by obtain rfl | ha := eq_or_ne a 0 <;> simp [*]
+
+/-- Equality holds when `a ≠ 0`. See `inv_mul_cancel`. -/
+lemma inv_mul_le_one : a⁻¹ * a ≤ 1 := by obtain rfl | ha := eq_or_ne a 0 <;> simp [*]
+
+/-- Equality holds when `a ≠ 0`. See `mul_inv_cancel_left`. -/
+lemma mul_inv_left_le (hb : 0 ≤ b) : a * (a⁻¹ * b) ≤ b := by
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Field/Defs.lean:48]: Could you either add the `nonpos` versions, or add a comment explaining why only the nonneg versions are present? 

bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**